src/OpenTelemetry.Instrumentation.Http/Implementation/TelemetryHelper.cs

```diff
@@ -0,0 +1,111 @@
+// <copyright file="TelemetryHelper.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Net;
+
+namespace OpenTelemetry.Instrumentation.Http.Implementation;
+
+internal static class TelemetryHelper
+{
+#pragma warning disable SA1509 // Opening braces should not be preceded by blank line
+    // Status Codes listed at http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
+    private static readonly Dictionary<HttpStatusCode, object> BoxedStatusCodes = new()
```

Comment:
Is Dictionary the best option here?

Reply:
What you have in mind as an alternative?

If you are asking if it should be `int` or `HttpStatusCode` key, actually seems faster to use the enum:

|                  Method |     Mean |     Error |    StdDev | Allocated |
|------------------------ |---------:|----------:|----------:|----------:|
| IntegerDictionaryLookup | 3.959 ns | 0.0560 ns | 0.0523 ns |         - |
|    EnumDictionaryLookup | 3.140 ns | 0.0399 ns | 0.0333 ns |         - |

<details>
<summary>Benchmark code</summary>

```csharp
using System.Net;
using BenchmarkDotNet.Attributes;

namespace OpenTelemetry.Benchmarks;

public class EnumDictionaryBenchmarks
{
    private readonly Dictionary<int, object> boxedStatusCodesWithIntKey = new()
    {
        [200] = 200,
    };

    private readonly Dictionary<HttpStatusCode, object> boxedStatusCodesWithHttpStatusCodeKey = new()
    {
        [HttpStatusCode.OK] = 200,
    };

    [Benchmark]
    public object IntegerDictionaryLookup()
    {
        return this.GetBoxedStatusCode((int)HttpStatusCode.OK);
    }

    [Benchmark]
    public object EnumDictionaryLookup()
    {
        return this.GetBoxedStatusCode(HttpStatusCode.OK);
    }

    private object GetBoxedStatusCode(int statusCode)
    {
        if (this.boxedStatusCodesWithIntKey.TryGetValue(statusCode, out object boxedStatusCode))
        {
            return boxedStatusCode;
        }

        return statusCode;
    }

    private object GetBoxedStatusCode(HttpStatusCode statusCode)
    {
        if (this.boxedStatusCodesWithHttpStatusCodeKey.TryGetValue(statusCode, out object boxedStatusCode))
        {
            return boxedStatusCode;
        }

        return (int)statusCode;
    }
}
```
</details>

Reply:
Is there a faster version? (dictionary might be doing some hashing or tree search, here we only need a `O(1)` lookup, an array might even work better? (or frozen dict))

Reply:
I pushed an array version. It is fast:

|                  Method |      Mean |     Error |    StdDev | Allocated |
|------------------------ |----------:|----------:|----------:|----------:|
| IntegerDictionaryLookup | 4.0332 ns | 0.0862 ns | 0.0806 ns |         - |
|    EnumDictionaryLookup | 3.2120 ns | 0.0694 ns | 0.0649 ns |         - |
|         EnumArrayLookup | 0.7365 ns | 0.0336 ns | 0.0314 ns |         - |

I put in some logic so we don't have to keep 0-100 non-valid status codes in the array. That saves some memory at the cost of a bit of speed. How you feel about that tradeoff?

Reply:
Either way looks good to me.